### PR TITLE
Add a troubleshooting doc

### PR DESCRIPTION
**Pull request by henrydavidge**
Signed-off-by: Henry D <henrydavidge@gmail.com>

## What changes are proposed in this pull request?
We can add to this as we find new common issues.

## How is this patch tested?
- [ ] Unit tests
- [ ] Integration tests
- [x] Manual tests

(Details)

### CODE REVIEW

File: docs/source/troubleshooting.rst [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+===============
+
+- Job is slow or OOMs while using an aggregate like ``collect_list`` or
+  ``sample_call_summary_stats``
+
+  * Try disabling the `ObjectHashAggregate
+    <https://github.com/apache/spark/commit/27daf6bcde782ed3e0f0d951c90c8040fd47e985>`_ by setting
+    ``spark.sql.execution.useObjectHashAggregateExec`` to ``false``
+
+- Job is slow or OOMs while writing to partitioned table
```
Comment (kianfar77) [inline docs/source/troubleshooting.rst]: nit: the line spacing is awkward on rendering. The solution is closer to the next item than the item.

File: docs/source/troubleshooting.rst [hunk 2]
```diff
@@ -0,0 +1,20 @@
+Troubleshooting
+===============
+
+- Job is slow or OOMs while using an aggregate like ``collect_list`` or
```
Comment (kianfar77) [inline docs/source/troubleshooting.rst]: nit: perhaps define OOM once.

File: docs/source/troubleshooting.rst (hunk 1 above)
Comment (henrydavidge) [reply docs/source/troubleshooting.rst]: Fixed

Comment (henrydavidge) [reply docs/source/troubleshooting.rst]: Fixed by the css change

Comment (kianfar77) [reply docs/source/troubleshooting.rst]: Nice

**Pull request merged**